### PR TITLE
restruct compressed_tensors_w8a8_fp8

### PR DESCRIPTION
**Pull request by BBuf**
## Motivation

- [x] Merge the `Fp8LinearOp` class with the `apply_fp8_linear` function to reduce redundant repetitive code.
- [x] Add compressed-tensors w8a8fp8 format model test in ci.  

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

File: python/sglang/srt/layers/quantization/fp8_utils.py [hunk 1]
```diff
@@ -232,218 +233,87 @@ def channel_quant_to_tensor_quant(
...
+    input_2d_shape,
+    output_shape,
+    bias,
+    input_dtype,
+):
+    global TORCH_DEVICE_IDENTITY
+    if TORCH_DEVICE_IDENTITY is None:
+        TORCH_DEVICE_IDENTITY = torch.ones(1, dtype=torch.float32)
+    if TORCH_DEVICE_IDENTITY.device != weight.device:
+        TORCH_DEVICE_IDENTITY = TORCH_DEVICE_IDENTITY.to(weight.device)
```
Comment (merrymercy) [inline python/sglang/srt/layers/quantization/fp8_utils.py]:
```suggestion
    if TORCH_DEVICE_IDENTITY is None:
        TORCH_DEVICE_IDENTITY = torch.ones(
            1, dtype=torch.float32, device=weight.device
        )
```

File: python/sglang/srt/layers/quantization/fp8_utils.py [hunk 2]
```diff
@@ -232,218 +233,87 @@ def channel_quant_to_tensor_quant(
...
+    pad_output: Optional[bool] = None,
+    compressed_tensor_quant: bool = False,
 ) -> torch.Tensor:
+    # Note: we pad the input because torch._scaled_mm is more performant
+    # for matrices with batch dimension > 16.
+    # This could change in the future.
+    # We also don't pad when using torch.compile,
+    # as it breaks with dynamic shapes.
+    if pad_output is None:
+        enable_torch_compile = os.environ.get(
```
Comment (merrymercy) [inline python/sglang/srt/layers/quantization/fp8_utils.py]: use get_bool_env_var

File: python/sglang/srt/layers/quantization/fp8_utils.py (hunk 1 above)
Comment (BBuf) [reply python/sglang/srt/layers/quantization/fp8_utils.py]: done

File: python/sglang/srt/layers/quantization/fp8_utils.py (hunk 2 above)
Comment (BBuf) [reply python/sglang/srt/layers/quantization/fp8_utils.py]: done.

**Review by merrymercy — APPROVED** [review]
